interact/userio.go

```diff
@@ -27,6 +28,14 @@ func newTTYUser(input io.Reader, output io.Writer) ttyUser {
 	}
 }
 
+func (u ttyUser) detectTerminalSize(file *os.File) error {
```

Comment:
could you move this logic into the constructor instead by having the constructor take the `file`? i don't like that this private method is alongside other public methods, but only ever called by the person constructing the object.